src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -956,27 +963,37 @@ namespace winrt::TerminalApp::implementation
         winrt::hstring searchText{ _getTrimmedInput() };
 
         auto commandsToFilter = _commandsToFilter();
-        for (const auto& action : commandsToFilter)
-        {
-            // Update filter for all commands
-            // This will modify the highlighting but will also lead to re-computation of weight (and consequently sorting).
-            // Pay attention that it already updates the highlighting in the UI
-            action.UpdateFilter(searchText);
 
-            // if there is active search we skip commands with 0 weight
-            if (searchText.empty() || action.Weight() > 0)
+        if (_currentMode == CommandPaletteMode::TabSwitchMode)
+        {
+            for (uint32_t index = 0; index < commandsToFilter.Size(); index++)
+            {
+                actions.push_back(commandsToFilter.GetAt(index));
+            }
```

Comment:
nit: is this transformable to a `std::copy` with a `back_inserter`? I remember Carlos and I had a similar discussion on the Settings editor UI

Reply:
@DHowett - semantically, yes :blush: Technically, I am not sure how to do it because commandsToFilter is an IVector. Is there a way to get the underlying vector / iterator for it?

Reply:
This works, but I have no idea why. VS won't tell me exactly where `begin` and `end` originate.

```cpp
	std::vector<winrt::hstring> vec{ L"One", L"Two", L"Three" };
	auto f = winrt::single_threaded_vector(std::move(vec));
	Collections::IVector<winrt::hstring> projectedVector{ f };

	std::for_each(begin(projectedVector), end(projectedVector), [](auto&& val) {
		std::wcout << std::wstring_view{ val } << L"\n";
	});
```

I'm not `using namespace winrt::impl`, but here we are in 2021 and all signs point to that I am using begin/end directly out of `winrt::impl`.

Reply:
This is absolutely cool! Is it documented somewhere? :smile:

Reply:
As far as I can tell, it's a language feature (!) that in part is used to support range-based for loops. I've been digging for documentation (and trying to figure out how it works), but it has been difficult.

Reply:
C:\Users\vikhait\Documents\GitHub\terminal\src\cascadia\TerminalApp\Generated Files\winrt\Windows.Foundation.Collections.h

```
template <typename T, std::enable_if_t<has_GetAt<T>::value, int> = 0>
    fast_iterator<T> begin(T const& collection) noexcept
    {
        return { collection, 0 };
    }

    template <typename T, std::enable_if_t<has_GetAt<T>::value, int> = 0>
    fast_iterator<T> end(T const& collection)
    {
        return { collection, collection.Size() };
    }
```

Reply:
Right, but if nobody is `using namespace winrt::impl` (which they tell us is private :wink:), those definitions _shouldn't be found_ when they're not namespace-qualified. Yet this works.

Reply:
Got it! This is clever. Argument-dependent lookup (ADL) specifies how free functions are resolved (based on their argument types). One of the steps in argument-dependent lookup is the collection of the class hierarchy and the namespaces of the classes in that hierarchy.

Because each C++/WinRT projection type inherits from something in `winrt::impl`, ADL on projection types finds any free functions inside the `impl` namespace. If they didn't inherit from `winrt::impl`, `winrt::impl::begin` would not be found.

Reply:
This is really clever.. but it also makes the modern c++ extremely challenging to code review IMHO.